src/inter-aut.cc

```diff
@@ -0,0 +1,410 @@
+//
+// Created by Martin Hruska on 10.08.2022.
+//
+
+#include <mata/inter-aut.hh>
+#include <mata/util.hh>
+
+namespace
+{
+    bool is_logical_operator(char ch)
+    {
+        return (Mata::util::haskey(std::set<char>{'&', '|', '!'}, ch));
+    }
+
+    Mata::InterAutomaton::Naming get_naming_type(const std::string &key)
+    {
+        assert(key.find('-') != std::string::npos);
+        const std::string& type = key.substr(key.find('-')+1, std::string::npos);
+
+        if (type == "auto")
+            return Mata::InterAutomaton::Naming::AUTO;
+        else if (type == "enum")
+            return Mata::InterAutomaton::Naming::ENUM;
+        else if (type == "marked")
+            return Mata::InterAutomaton::Naming::MARKER;
+
+        assert(false);
```

Comment:
Why do you think that an else branch is better? I like this way because it is obvious that the function should not end there.

Reply:
By all means, do not add an `else` branch, but at least give the user who will fail on this assert some useful error message.

```cpp
assert(false && "Should never reach here because ...");
```

Reply:
done